blocks/loops.js

```diff
@@ -327,24 +327,23 @@ Blockly.Constants.Loops.CONTROL_FLOW_IN_LOOP_CHECK_MIXIN = {
   /**
    * Called whenever anything on the workspace changes.
    * Add warning if this flow block is not nested inside a loop.
-   * @param {!Blockly.Events.Abstract} _e Change event.
+   * @param {!Blockly.Events.Abstract} e Change event.
    * @this {Blockly.Block}
    */
-  onchange: function(_e) {
-    if (!this.workspace.isDragging || this.workspace.isDragging()) {
+  onchange: function(e) {
+    if (!this.workspace.isDragging || this.workspace.isDragging() ||
+        e.type != Blockly.Events.BLOCK_MOVE || e.blockId != this.id) {
```

Comment:
This updates it to:
- don't change state at the start of a drag
- don't change state if it's not a move
- don't change state if the moving block isn't this block

Reply:
Update the comment--but otherwise looks good.